internal/apps/dop/providers/search/handlers/issue/issue.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package issue
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"google.golang.org/protobuf/types/known/structpb"
+
+	issuepb "github.com/erda-project/erda-proto-go/dop/issue/core/pb"
+	"github.com/erda-project/erda-proto-go/dop/search/pb"
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/bundle"
+	"github.com/erda-project/erda/internal/apps/dop/providers/issue/core/query"
+	"github.com/erda-project/erda/internal/apps/dop/providers/search/handlers"
+)
+
+type issueHandler struct {
+	handlers.BaseHandler
+
+	uiPublicURL string
+	bdl         *bundle.Bundle
+	query       query.Interface
+}
+
+var (
+	projectLimit = 50
+)
+
+type Option func(*issueHandler)
+
+func WithQuery(query query.Interface) Option {
+	return func(i *issueHandler) {
+		i.query = query
+	}
+}
+
+func WithBundle(bdl *bundle.Bundle) Option {
+	return func(i *issueHandler) {
+		i.bdl = bdl
+	}
+}
+
+func WithUIPublicURL(uiPublicUrl string) Option {
+	return func(i *issueHandler) {
+		i.uiPublicURL = uiPublicUrl
+	}
+}
+
+func WithNexts(nexts ...handlers.Handler) Option {
+	return func(i *issueHandler) {
+		i.Nexts = nexts
+	}
+}
+
+func (i *issueHandler) DoSearch(ctx context.Context, req *pb.SearchRequest) {
+	defer i.DoNexts(ctx, req)
```

Comment:
Unintuitive func invoke here in defer.
If error happened, invoke `DoNexts` is necessary or not?

Reply:
`DoNexts` is necessary, the current search should not affect others